packages/backend/src/api/jobScraping/scraping.ts

```diff
@@ -0,0 +1,353 @@
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+
+  page.goto(smartRecruiters);
+
+  await page.waitForSelector('.jobs-item');
+
+  const jobLinks: any[] = [];
+
+  const jobListings = await page.$$('.jobs-item');
+  for (const link of jobListings) {
+    const href = await link.$eval('a', (el) => el.href);
+    jobLinks.push(href);
+  }
+
+  const jobs: any[] = [];
+
+  for (const link of jobLinks) {
+    await page.goto(jobLinks[1]);
+    const html = await page.content();
+    const $ = cheerio.load(html);
+
+    const title = $('.job-title').text().trim();
+    const description = $(
+      '#st-companyDescription > div:nth-child(2) > p , #st-jobDescription > div:nth-child(2) > p, #st-jobDescription > div:nth-child(2) > ul:nth-child(3) > li , #st-additionalInformation > div:nth-child(2) > ul:nth-child(2) > li'
+    )
+      .text()
+      .trim();
+    const { languages, frameworks } = filterLanguagesAndFrameworks(description);
+
+    const applicationLink = await shortenLink(link);
+
+    jobs.push({ title, languages, frameworks, applicationLink });
+  }
+
+  return jobs;
+};
+
+export const baytScrape = async () => {
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+
+  page.goto(bayt);
+
+  await page.waitForSelector('#results_inner_card');
+
+  const jobLinks: any[] = [];
+
+  const jobListings = await page.$$('.has-pointer-d');
+  for (const link of jobListings) {
+    const href = await link.$eval('a', (el) => el.href);
+    jobLinks.push(href);
+  }
+
+  const jobs: any[] = [];
+
+  for (const link of jobLinks) {
+    await page.goto(link);
+    const html = await page.content();
+    const $ = cheerio.load(html);
+
+    const title = $('#job_title').text().trim();
+    const description = $('div.t-break').text().trim();
+
+    const { languages, frameworks } = filterLanguagesAndFrameworks(description);
+
+    const applicationLink = await shortenLink(link);
+
+    jobs.push({ title, languages, frameworks, applicationLink });
+  }
+  return jobs;
+};
+
+//TODO:fix filtering the locations to get the remote jobs only
+export const remoteSourceScrape = async () => {
+  const browser = await puppeteer.launch();
+  const page = await browser.newPage();
+
+  page.goto(remoteSource);
+
+  await page.waitForSelector('.grouped-job-result');
+
+  const jobListings = await page.$$('.grouped-job-result');
+  console.log(`Found ${jobListings.length} job listings`);
+
+  for (const job of jobListings) {
+    const location = await job.evaluate(() => {
+      return document.querySelector('.job-list-company-meta-item.job-list-company-meta-locations')?.textContent;
+    });
+
+    console.log(`Location: ${location}`);
+
+    if (location?.split('-').includes('Remote')) {
+      console.log('Location is remote');
+
+      await page.click('div.grouped-job-result:nth-child(4) > button:nth-child(4)');
+
+      const title = await page.$eval(
```

Comment:
description, title, and link, are 3 separate async fetches... we don't need to await each synchronously.

checkout Promise.all() this will optimize the awaits